src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/GoGoDuckModule.java

```diff
@@ -37,104 +46,126 @@ public URIList getUriList() throws GoGoDuckException {
-            logger.warn("Could not find 'title' attribute in result file");
+            File tmpFile = File.createTempFile("ncpdq", ".nc");
+
+            List<String> command = new ArrayList<String>();
+            command.add(GoGoDuckConfig.ncpdqPath);
+            command.add("-O");
+            command.add("-U");
+            command.add(file.getAbsolutePath());
+            command.add(tmpFile.getAbsolutePath());
+
+            logger.info(String.format("Unpacking file (ncpdq) '%s' to '%s'", file.toPath(), tmpFile.toPath()));
+            GoGoDuck.execute(command);
+
+            Files.delete(file.toPath());
+            Files.move(tmpFile.toPath(), file.toPath());
+        } catch (Exception e) {
+            throw new GoGoDuckException(String.format("Could not run ncpdq on file '%s'", file.toPath()));
         }
+    }
 
-        newAttributeList.add(new Attribute("title",
-                String.format("%s, %s, %s",
-                        title,
-                        subset.get("TIME").start,
-                        subset.get("TIME").end)));
-
-        newAttributeList.add(new Attribute("geospatial_lat_min", subset.get("LATITUDE").start));
-        newAttributeList.add(new Attribute("geospatial_lat_max", subset.get("LATITUDE").end));
-
-        newAttributeList.add(new Attribute("geospatial_lon_min", subset.get("LONGITUDE").start));
-        newAttributeList.add(new Attribute("geospatial_lon_max", subset.get("LONGITUDE").end));
-
-        newAttributeList.add(new Attribute("time_coverage_start", subset.get("TIME").start));
-        newAttributeList.add(new Attribute("time_coverage_end", subset.get("TIME").end));
+    public List<String> ncksExtraParameters() {
+        List<String> ncksExtraParameters = new ArrayList<String>();
 
-        return newAttributeList;
+        Object layerKey = GoGoDuckConfig.getPropertyKeyByValuePart(profile);
+        if (layerKey != null) {
+            String layer = (String) layerKey;
+            String ncksParametersProperty = String.format("%s.ncks.parameters", layer);
+            String ncksParameters[] = GoGoDuckConfig.properties.getProperty(ncksParametersProperty).split(";", -1);
+            for (String ncksParameter : ncksParameters) {
+                ncksExtraParameters.add(ncksParameter);
+            }
+        }
+        return ncksExtraParameters;
     }
 
     public final void updateMetadata(Path outputFile) {
         try {
-            NetcdfFileWriter nc = NetcdfFileWriter.openExisting(outputFile.toAbsolutePath().toString());
+            String location = outputFile.toAbsolutePath().toString();
+            NetcdfFileWriter nc = NetcdfFileWriter.openExisting(location);
+            GridDataset gridDs = GridDataset.open(location);
 
             nc.setRedefineMode(true);
-            for (Attribute newAttr : getGlobalAttributesToUpdate(nc.getNetcdfFile())) {
+            for (Attribute newAttr : gridDs.getGlobalAttributes()) {
```

Comment:
Probably - need to check with PO's - there needs to be a record of the subset that was applied in the dataset